src/services/completions.ts

```diff
@@ -1635,13 +1635,17 @@ namespace ts.Completions {
         // invalid identifier name. We need to check if whatever was inside the quotes is actually a valid identifier name.
         // e.g "b a" is valid quoted name but when we strip off the quotes, it is invalid.
         // We, thus, need to check if whatever was inside the quotes is actually a valid identifier name.
-        if (performCharacterChecks && !isIdentifierText(name, target)) {
-            return undefined;
+        if (performCharacterChecks && !isIdentifierText(name, target) && !isNumericLiteralText(name)) {
+            return JSON.stringify(name);
         }
 
         return name;
     }
 
+    function isNumericLiteralText(name: string): boolean {
+        return /^-?\d+(\.\d+)?(e\d+)?$/.test(name);

```

Comment:
There are contexts like property names:
```ts
            const foob = {
                -0() {
                    return 1;
                }
            };
```
 where a leading minus would never be allowed (only an identifier, string literal, or number) - I don't think this should include the check for `-?`?

Reply:
Also, this helper probably should be somewhere central, since it will need to be updated once we support [the numeric seperator proposal](https://github.com/tc39/proposal-numeric-separator).

Reply:
We can just use a string literal in this case then, which makes this function unnecessary.